more_itertools/recipes.py

```diff
@@ -368,6 +374,11 @@ def grouper(iterable, n, incomplete='fill', fillvalue=None):
         return _zip_equal(*args)
     if incomplete == 'ignore':
         return zip(*args)
+    if incomplete == 'allow':
+        return (
+            takewhile(lambda x: x is not _marker, group)
+            for group in zip_longest(*args, fillvalue=_marker)
+        )
```

Comment:
I'm not sure about return type differences from other modes.
Should it be as following?
```python
return (
    tuple(takewhile(...)) ...
)
```